### PR TITLE
Add test for TraceManager.

### PR DESCRIPTION
**Pull request by markdewing**
It appears there exists test code to check compilation in TraceManager.cpp.
Move that to a unit test and fix it so it compiles and runs.

The operations in the first test may not be called in the correct order and may not even be consistent.   They do run without crashing, though.  
I would appreciate input on what the correct calling sequence should be.

### CODE REVIEW

File: src/Estimators/tests/CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,10 @@ SET(UTEST_NAME unit_test_${SRC_DIR})
 
 SET(SRCS test_accumulator.cpp test_local_energy_est.cpp test_manager.cpp)
 
+IF (NOT REMOVE_TRACEMANAGE)
```
Comment (prckent) [inline src/Estimators/tests/CMakeLists.txt]: What is the concern about this dependency? Also, REMOVE_TRACEMANAGER not TRACEMANAGE,

Comment (markdewing) [reply src/Estimators/tests/CMakeLists.txt]: It was to match some conditionals in the code. 
Now i tested compilation with REMOVE_TRACEMANAGER, and the unit test compiles and runs okay.  I'll remove the conditional.

**Comment by grahamlopez** [comment]
test this please

**Comment by grahamlopez** [comment]
test this please

**Comment by grahamlopez** [comment]
test this please

**Comment by grahamlopez** [comment]
test this please

**Comment by grahamlopez** [comment]
test this please

**Pull request merged**